src/placeos-models/settings.cr

```diff
@@ -77,10 +77,12 @@ module PlaceOS::Model
     validate ->(this : Settings) do
       if this.settings_string_changed?
         unencrypted = Encryption.is_encrypted?(this.settings_string) ? this.decrypt : this.settings_string
-        begin
-          YAML.parse(unencrypted) rescue JSON.parse(unencrypted)
-        rescue
-          this.validation_error(:settings_string, "is invalid JSON/YAML")
+        unless unencrypted.strip.empty?
+          begin
+            YAML.parse(unencrypted).as_h rescue JSON.parse(unencrypted)
+          rescue
+            this.validation_error(:settings_string, "is invalid JSON/YAML")
+          end
```

Comment:
I think this block might be best if refactored to be clear on what's invalid about the JSON.